robot/framesystem/framesystem.go

```diff
@@ -228,19 +228,26 @@ func (svc *frameSystemService) updateLocalParts(ctx context.Context) error {
 		if c.Frame.Parent == "" {
 			return errors.Errorf("parent field in frame config for part %q is empty", c.Name)
 		}
-		if c.Frame.ID == "" {
-			c.Frame.ID = c.Name
+		cfgCopy := &referenceframe.LinkConfig{
+			ID:          c.Frame.ID,
+			Translation: c.Frame.Translation,
+			Orientation: c.Frame.Orientation,
+			Geometry:    c.Frame.Geometry,
```

Comment:
Is there any chance these will ever get modified? Because they're also pointers, so you're still pointing to the structs in the original config here. If so, then a deep copy may be needed.

Reply:
No, except inside this very function. The only thing we do with the struct is `ParseConfig`, which resolves all the pointer config objects into new structs.